include/sparrow/typed_array.hpp

```diff
@@ -186,7 +194,31 @@ namespace sparrow
         layout_type m_layout;
     };
 
+    template <typename T, std::size_t N>
+struct template_param_at;
```

Comment:
This is essentially `mpl::at_t<...>` algorithm that I didnt implement in the end because I didnt need it finally.
Could you rename it and move it in mp_utils.hpp?

As for the implementation, I wanted to avoid recursion as it's a major source of compilation time and we seem to have some other ways in C++20 to do it that would be better but I didnt have time to get into the details yet.

Reply:
Finally I don't need it, I removed it